trino/client.py

```diff
@@ -747,7 +748,14 @@ def execute(self, additional_http_headers=None) -> TrinoResult:
         if self.cancelled:
             raise exceptions.TrinoUserError("Query has been cancelled", self.query_id)
 
-        response = self._request.post(self._sql, additional_http_headers)
+        if self._experimental_python_types:
```

Comment:
Actually you can always enable `PARAMETRIC_DATETIME`. This is also how the JDBC driver works. Please also adjust the commit message and PR description.

Reply:
Shouldn't this be part of another PR?

Reply:
Done